lib/rules/selector-no-qualifying-type/index.mjs

```diff
@@ -22,15 +24,148 @@ const meta = {
 
 const HAS_ID_CLASS_ATTRIBUTE_PREFIXES = /[#.[]/;
 
-/** @import { Tag, Node, Selector, Container } from 'postcss-selector-parser' */
+const IS_SELECTOR_CONTAINING_PSEUDO = /^:(?:has|host|is|matches|not|where)$/i;
```

Comment:
[suggestion] What about the following refactoring?

- Utilize the existing `logicalCombinationsPseudoClasses`
  - Ref https://github.com/stylelint/stylelint/blob/afe57f9a524aa5a8106521ae65543ec47b7f033e/lib/reference/selectors.mjs#L302
- Use `Set` instead of Regexp for readability
- Create a private function to reduce code duplication

E.g.

```js
import { logicalCombinationsPseudoClasses } from '../../reference/selectors.mjs';

const FUNCTIONAL_PSEUDO_CLASSES = new Set(
	[...logicalCombinationsPseudoClasses, 'host'].map((s) => `:${s}`),
);

/**
 * @param {unknown} node
 * @returns {node is selectorParser.Pseudo}
 */
function isFunctionalPseudoClass(node) {
	return selectorParser.isPseudoClass(node) && FUNCTIONAL_PSEUDO_CLASSES.has(node.value);
}
```

Reply:
`:lang` is also a functional pseudo class, as the functional part of the name implies the `()` in contrast to for example `:hover`, which is a plain pseudo class.

I don't think there is an existing name to group all functional pseudo classes that accept a selector (list) as their argument.

Otherwise this suggested change is fine with me, but I would rather keep a descriptive name to indicate that this list includes things like `:host(.foo)` but not `:dir(ltr)`

Open to suggestions for a better name :)